### DISCUSSION TITLE
Training time for RL in simulators

### DISCUSSION THREAD

**Discussion by paulpacaud** [RLinf/RLinf#481] [Q&A] [answered]
Hi community,
I was testing the RLinf library on the example "RL with Metaworld Benchmark':
- model: pi0, algo: PPO, train/eval: MT50, I did not touch the default config: examples/embodiment/config/metaworld_50_ppo_openpi.yaml

It took 30min per epoch on one H100  (80Gb RAM, 24 CPUs), and the default value in the config.yaml is to run it for 1000 epochs! It seems a lot, it would take 30min*1000=500 hours to fine-tune pi0 with PPO on one H100!

Is this order of magnitude aligned with what you also observed?
Is it so long only on Metaworld, or only for pi0? Are there simulators or models providing faster training? How do you speed up your training?

Happy to discuss more together

**Comment by chenkang455** [comment, marked as answer]
Hi @paulpacaud, sorry for the delay. I’ll address your questions one by one.

First, let me introduce our experimental setup: we conducted tests on 8x H100 GPUs, loading 8 environments per GPU and running for 8 times, totaling 512 environments per RL training step. Each rollout takes approximately 8 minutes. We set the maximum steps to 1,000, but we don't always run to the end; we monitor the convergence of the curves and terminate manually—for MetaWorld, we stop around 400 rounds.

Regarding RL training time, convergence speeds vary across tasks. For instance, pi0 typically converges within 100 steps on LIBERO-Spatial, which takes about 8 hours on 8x H100s. If you have limited hardware, we have recently identified a set of hyperparameters that allow training with fewer environments. Check here: https://github.com/RLinf/RLinf/issues/470#issuecomment-3680721478. While the performance might not exceed our current configuration, our results on LIBERO-Spatial show it still yields improvements. You can refer to this config and give it a try.


Besides, we have include a quick start yaml for pi0 model, which requires less gpu memory on libero spatial task, check here https://github.com/RLinf/RLinf/pull/493 for details.

**Comment by chenkang455** [comment]
Check our training curve on pi0 libero here:

<img width="2158" height="678" alt="image" src="https://github.com/user-attachments/assets/e5443b32-ea0c-4346-b1d5-fe1e997cf081" />

**Comment by paulpacaud** [comment]
Hi, just giving some feedback on the suggested configuration for finetuning pi0.5 with PPO on Metaworld 50 with low-resource:
```
rollout_epoch: 8 -> 2
total_num_envs: 64 -> 32
micro_batch_size: 128 -> 64
global_batch_size: 2048 -> 256
lr: 5e-6 -> 1e-6
critic_warmup_steps: 0 # optional 0,5,10, if you observe performance drop at first
```

I obtain these results, which are lower than the ones reported in the official doc but still shows improvement of pi0.5 

Source | Model | Reward | Easy | Medium | Hard | Very Hard | Avg.
-- | -- | -- | -- | -- | -- | -- | --
reproduced | π0.5 | - | 66.4 | 42.7 | 31.7 | 32 | 43.2
reproduced | π0.5 + PPO | simulator sparse | 83.9 | 50 | 58.3 | 64 | 64.1
paper | π0.5 | - | 68.2 | 37.3 | 41.7 | 28 | 43.8
paper | π0.5 + PPO | simulator sparse | 86.4 | 55.5 | 75 | 66 | 70.7

I stop at 600 global steps, training longer did not help.
<img width="1068" height="994" alt="image" src="https://github.com/user-attachments/assets/d8d8cc32-7540-4f0b-bb7e-1b52bd3c835f" />

Do you find these results and curves aligned with your results?

**Reply by chenkang455** [reply]
Hi @paulpacaud, thanks for the feedback.

I haven’t evaluated this minimal config on MetaWorld; my numbers come from LIBERO-Spatial, where the converged success rate drops slightly from 98 % (full config) to 92 % (minimal).

For MetaWorld we simply halted the original training run at 400 k steps and have uploaded the checkpoint to Hugging Face.

Besides, π₀ also appears to work better.

**Reply by paulpacaud** [reply]
Thanks for the reply, although not on the same simulator, the drop you report and the one I observe seem to be aligned.
